cloud-common/src/main/java/com/sequenceiq/cloudbreak/polling/ExtendedPollingResult.java

```diff
@@ -44,9 +47,15 @@ public Set<Long> getFailedInstanceIds() {
         return failedInstanceIds;
     }
 
+    public Set<String> getFailedHostNames() {
```

Comment:
This is not a good addition to the ExtendedPollingResult class, especially if there is going to be a discrepancy between what is returned by getFailedHostNames, and getFailedInstanceIds (which is the case here). Becomes a potential debugging nightmare if someone uses this in the future, and (would typically) assume that getFailedHostNames and  getFailedInstanceIds represent the same set of hosts.

Even with Javadocs calling this out, I'd be very hesitant to make this change.

How expensive would it be to translate to instanceIds and back ?